api/src/main/java/jakarta/persistence/PessimisticLockScope.java

```diff
@@ -11,53 +11,95 @@
  */
 
 // Contributors:
+//     Gavin King      - 4.0
 //     Linda DeMichiel - 2.1
 //     Linda DeMichiel - 2.0
 
 package jakarta.persistence;
 
 /**
- *
- * Defines the values of the {@code jakarta.persistence.lock.scope}
- * property for pessimistic locking.  This property may be passed as an
- * argument to the methods of the {@link EntityManager}, {@link Query},
- * and {@link TypedQuery} interfaces that allow lock modes to be specified
- * or used with the {@link NamedQuery} annotation.
+ * Controls how a {@linkplain LockModeType pessimistic lock} applied to
+ * an entity affects associated collections and relationships.
+ * <p>
+ * A {@code PessimisticLockScope} may be passed as a {@link FindOption}
+ * {@link RefreshOption}, or {@link LockOption}. Alternatively, it may
+ * be specified via the property {@code jakarta.persistence.lock.scope}.
+ * This property may be passed as an argument to methods of
+ * {@link EntityManager}, {@link Query}, and {@link TypedQuery} which
+ * accept a {@link LockModeType} or it may be used in the {@link NamedQuery}
```

Comment:
This statement seems partially wrong.
I see that EntityManager has methods that accept LockModeType along with properties that can include this.
But the only methods I see on Query and TypedQuery that accept LockModeType are,
```
setLockMode(LockModeType lockMode)
```
which does not have a properties argument that could take a `jakarta.persistence.lock.scope`

Reply:
It's called `setHint()`, and it's documented as:

> Set a query **property** or hint.

Yeah, don't look at me.